tonilopezmr/day 2/ChecksumPartOne.kt

```diff
@@ -0,0 +1,71 @@
+package day2
+
+import org.junit.Assert.*
+import org.junit.Test
+
+class ChecksumPartOne {
+
+  fun checksum(of: Array<Array<Int>>): Int {
+    return of.fold(emptyList<Int>()) { acc, a ->
+      acc.plus(maxMinusMin(a))
+    }.reduce { acc, i -> acc + i }
+  }
+
+  fun maxMinusMin(array: Array<Int>): Int {
+    var min = array[0]
+    var max = 0
```

Comment:
Mutabiliy??!?
![](https://media.giphy.com/media/KupdfnqWwV7J6/giphy-tumblr.gif)

Just kidding, nice solution. Just a question, why didn't you used `max` and `min` methods of Kotlin?

Reply:
I didn't want to do the mistake I did in the last comment